src/features/jserrors/aggregate/compute-stack-trace.js

```diff
@@ -66,9 +67,34 @@ var gecko = /^\s*(?:(\S*|global code)(?:\(.*?\))?@)?((?:file|http|https|chrome|s
 var chrome_eval = /^\s*at .+ \(eval at \S+ \((?:(?:file|http|https):[^)]+)?\)(?:, [^:]*:\d+:\d+)?\)$/i
 var ie_eval = /^\s*at Function code \(Function code:\d+:\d+\)\s*/i
 
-export function computeStackTrace (ex) {
+/**
+ * Represents an error with a stack trace.
+ * @typedef {Object} StackInfo
+ * @property {string} name - The name of the error (e.g. 'TypeError').
+ * @property {string} message - The error message.
+ * @property {string} stackString - The stack trace as a string.
+ * @property {Array<Object>} frames - An array of frames in the stack trace.
+ * @property {string} frames.url - The URL of the file containing the code for the frame.
+ * @property {string} frames.func - The name of the function associated with the frame.
+ * @property {number} frames.line - The line number of the code in the frame.
+ */
+
+/**
+ * Used for comparison to decide if a stack trace URL is inline.
+ */
+var _loaderOriginUrl
+
+/**
+ * Attempts to compute a stack trace for the given exception.
+ * @param {Error} ex - The exception for which to compute the stack trace.
+ * @param {string} loaderOriginUrl - The origin URL of the agent loader, for identifying inline scripts.
+ * @returns {StackInfo} A stack trace object containing information about the frames on the stack.
+ */
+export function computeStackTrace (ex, loaderOriginUrl) {
   var stack = null
 
+  _loaderOriginUrl = loaderOriginUrl
```

Comment:
It feels like we are storing some flavor of the origin in a lot of places. This probably isn't a big deal, but just worth calling it out.

Reply:
Actually, it looks like `canonicalizeUrl` will call cleanUrl.  Can we just pass in the origin directly instead of keeping a "cleaned url" in the agg file and passing it around?

Reply:
Now getting current page location directly within canonicalizeUrl.